include/ada/href_aggregator.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file href_aggreator.h
+ * @brief Definitions for the href_aggregator.
+ */
+#ifndef ADA_HREF_AGGREGATOR_H
+#define ADA_HREF_AGGREGATOR_H
+
+#include "ada/encoding_type.h"
+#include "ada/expected.h"
+#include <optional>
+#include <string_view>
+
+/**
+ * @namespace ada::parser
+ * @brief Code for href_aggregator
+ */
+namespace ada::parser {
+
+/**
+ * Sometimes, you do not want to construct a full URL structure and you would
+ * be satisfied with just a normalized href string.
+ *
+ * The href_aggregator contains a single string (buffer) along with a
+ * set of offsets (components). That is sufficient to qualified as a fully
+ * parsed URL. However, unlike ada/url, updates can be more difficult because
+ * the components are not separate. However, we should be able to map an
+ * ada/href_aggregator to an ada/url quickly and efficiently.
+ *
+ * In effect, we can view ada/url and ada/href_aggregator as equivalent.
```

Comment:
I agree. I think we should create an abstract class from both implementations and extend from that before diving into the href::aggregator